src/hotspot/cpu/ppc/methodHandles_ppc.cpp

```diff
@@ -306,7 +313,10 @@ void MethodHandles::generate_method_handle_dispatch(MacroAssembler* _masm,
   if (receiver_reg != noreg)  assert_different_registers(temp1, temp2, temp3, temp4, receiver_reg);
   if (member_reg   != noreg)  assert_different_registers(temp1, temp2, temp3, temp4, member_reg);
 
-  if (iid == vmIntrinsics::_invokeBasic) {
+  if (iid == vmIntrinsics::_invokeBasic || iid == vmIntrinsics::_linkToNative) {
+    if (iid == vmIntrinsics::_linkToNative) {
+      assert(for_compiler_entry, "only compiler entry is supported");
+    }
```

Comment:
I believe it is customary to write conditional asserts like:
```
  assert(for_compiler_entry || iid != vmIntrinsics::_linkToNative, "only compiler entry is supported for linkToNative")
```

...same thing for S390.

Reply:
Thanks for reviewing it!
I've copied the code from x86 and I'd prefer to keep it consistent even though I like your version more. Are you ok with it?

Reply:
Yeah, that's fine. I don't have a strong opinion here.